platform/linux-generic/odp_ishm.c

```diff
@@ -909,22 +910,36 @@ static int do_unmap(void *start, uint64_t size, uint32_t flags,
  * Returns the index of the found block (if any) or -1 if none.
  * Mutex must be assured by the caller.
  */
-static int find_block_by_name(const char *name)
+static int find_block_by_name(const char *name, odp_bool_t use_param_name)
 {
 	int i;
 
 	if (name == NULL || name[0] == 0)
 		return -1;
 
 	for (i = 0; i < ISHM_MAX_NB_BLOCKS; i++) {
-		if ((ishm_tbl->block[i].len) &&
-		    (strcmp(name, ishm_tbl->block[i].name) == 0))
+		const char *block_name = use_param_name ?
+					 ishm_tbl->block[i].param_name :
+					 ishm_tbl->block[i].name;
+
+		if (ishm_tbl->block[i].len && strcmp(name, block_name) == 0)
 			return i;
 	}
 
 	return -1;
 }
 
+/*
+ * Make application given memory block name unique
+ */
+static void name_to_ishm_name(const char *name, char *block_name)
+{
+	int i = 0;
+
+	while (find_block_by_name(block_name, false) > 0)
```

Comment:
This looks racy if multiple threads happen to be reserving shm blocks at the same time? Maybe you need a mutex or maybe you should generate a unique name based on some global counter (which would need to be atomic and in shared memory)?

Reply:
There seems to be a lock there already (ishm_tbl->lock), so ignore this comment.